CONTRIBUTING.txt

```diff
@@ -0,0 +1,99 @@
+    2. All contributions to FreeCAD MUST use a compatible license.
+    3. All contributions are owned by their authors unless assigned to another.
+    4. FreeCAD does not have a mandatory copyright assignment policy.
+    5. A Contributor who wishes to be identified in the Credits section of the application "About" dialog is responsible for identifying themselves appropriately by notifying an Administrator or Maintainer.
+
+5. Contribution Requirements
+
+    1. Patches are submitted in the form of Pull Requests (PR).
+    2. Maintainers and Contributors MUST have a Platform account and SHOULD use their real names or a well-known alias. If the Platform alias differs from the account alias on the FreeCAD Forum, effort SHOULD be taken to avoid confusion.
+    3. A PR SHOULD be a minimal and accurate answer to exactly one identified and agreed-on problem.
+    4. A PR SHOULD refrain from adding additional dependencies to the FreeCAD project unless no other option is available.
+    5. Code submissions MUST adhere to the code style guidelines of the project if these are defined.
+    6. If a PR contains multiple commits, each commit MUST compile cleanly and add value to the history of the project. Checkpoint commits SHOULD be squashed.
+    7. A PR SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+    8. A PR MUST compile cleanly and pass project self-tests on all target Platforms.
+    9. Each commit message in a PR MUST succinctly explain what the commit achieves.
+    10. The PR Description MUST consist of a single short (less than 50 characters) line stating the problem (“Problem: …") being solved, followed by a blank line and then the proposed solution (“Solution: …").
+    11. A “Valid PR” is one which satisfies the above requirements.
+
+6. Process
+
+    1. Change on the project follows the pattern of accurately identifying problems and applying minimal, accurate solutions to these problems.
+    2. To request changes, a User (end user, Contributor, Maintainer, or Administrator) logs an issue on the project Platform issue tracker.
+    3. The User or Contributor SHOULD write the issue by describing the problem they face or observe. Links to the forum or other resources are permitted but the issue SHOULD be complete and accurate and SHOULD NOT require the reader to visit the forum or any other Platform to understand what is being described.
+    4. Issue authors SHOULD strive to describe the minimum acceptable condition.
+    5. Issue authors SHOULD focus on User tasks and avoid comparisons to other software solutions.
+    6. The User or Contributor SHOULD seek consensus on the accuracy of their observation and the value of solving the problem.
+    7. To submit a solution to a problem, a Contributor SHALL create a pull request back to the project.
+    8. Contributors and Maintainers SHALL NOT commit changes directly to the target branch.
```

Comment:
Shall this apply also for crash fixing ?

Reply:
8. Contributors and Maintainers SHALL NOT commit changes directly to the target branch.
I think I understand the intent of this item, but it doesn't make sense as written.  Eventually the commit has to be applied to the branch.   Also, the "contributor" role isn't supposed to commit anything to the project repo, only the "maintainer" role is supposed to do this.
suggestion: 
8. Maintainers shall not commit changes without an accompanying PR.

Reply:
Granting someone access to merge a PR also grants them access to push changes directly to the branch without a PR.  That's the behavior we're trying to curtail.  
In the case that a valid issue is followed by a correct PR with just one little typo, a maintainer might read the suggestion as, "Well there IS a PR and this is a minor change so it's ok".  
How about
    8.  Maintainers SHALL NOT push changes to the target branch without using the PR workflow.